test/tests/functional/pbs_array_job.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestArrayJobs(TestFunctional):
+
+    """
+    This test suite tests if PBS's array job functionality
+    """
+    def test_expired_subjobs_not_reported(self):
+        """
+        Test if a subjob is finished ans moves to expired state,
+        it is not reported to scheduler in the next scheduling
+        cycle. Scheduler expects only running subjobs to be reported to it.
+        """
+
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
```

Comment:
why do you need to set ncpus=1? looking at your test it doesn't matter if you have 1 or more.

Reply:
The reason is that I want some of the subjobs to be either in running state or queued state. This is why I made the available ncpus as 1 because then only one job can run at a time.

Reply:
okay but then this configuration would be hard to run on a CRAY or cpuset systems. why don't you submit an exclusive job so that PBS cannot run more than one job one a node?

Reply:
Well, you are right, but even if I take the node exclusively it might still not work because those systems come with loads of vnodes prebuilt. What I can probably do is that I can submit a job requesting mom name as the node where the job needs to run. This way it will only run on login node.